### PR TITLE
pythonPackages.pyvcd: init at 0.1.4

### PR DESCRIPTION
**Pull request by sbourdeauducq**
###### Motivation for this change
Add the pyvcd Python package required by some electronic design automation tools.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS) - Can't figure it out. Works fine with ``nix-shell --pure``.
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- N/A Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- N/A Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- N/A Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size: < 1MB
- N/A Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by dotlambda — CHANGES_REQUESTED** [review]
Adding yourself to `maintainer-list.nix` should be a separate commit. Also, prepend `pythonPackages.` to the existing commit's message.

**Comment by dotlambda** [comment]
> Ran 0 tests in 0.000s

Please add an appropriate `checkPhase` or - if there are no tests - set `doCheck = false` and add a comment with the reason.

File: pkgs/development/python-modules/pyvcd/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+buildPythonPackage rec {
+  version = "0.1.4";
+  pname = "pyvcd";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0dv9wac5y5z9j54ypyc59csxdiy9ybpphw9ipxp1k8nfg65q9jxx";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ setuptools_scm six ];
```
[the remark this answers is not in the document]
Comment (sbourdeauducq) [reply pkgs/development/python-modules/pyvcd/default.nix]: fixed

File: pkgs/development/python-modules/pyvcd/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{ lib, buildPythonPackage, fetchPypi, python3Packages }:
```
[the remark this answers is not in the document]
Comment (sbourdeauducq) [reply pkgs/development/python-modules/pyvcd/default.nix]: done

**Comment by sbourdeauducq** [comment]
> Please add an appropriate `checkPhase`

Done and tests passing. Thanks for the quick review!

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.pyvcd python3.pkgs.pyvcd

**Comment by sbourdeauducq** [comment]
@dotlambda Thanks for the merge!

@worldofpeace I understood this, but when I try to use ``nix-build`` I get a confusing error message. Likely I'm doing something dumb...

``` 
$ nix-build -I nixpkgs=. -p "python3.withPackages(ps: with ps; [ pyvcd ])"
these derivations will be built:
  /nix/store/57x6z6360valwawys7dw387w2wac1sqw-python3-3.7.2-env.drv
  /nix/store/l3qv1nlaavy6llv0k4b2863fz3mdqbgz-shell.drv
building '/nix/store/57x6z6360valwawys7dw387w2wac1sqw-python3-3.7.2-env.drv'...
created 245 symlinks in user environment
building '/nix/store/l3qv1nlaavy6llv0k4b2863fz3mdqbgz-shell.drv'...
builder for '/nix/store/l3qv1nlaavy6llv0k4b2863fz3mdqbgz-shell.drv' failed to produce output path '/nix/store/vx5y03618gf3hwn94fa2lyry3nmd40fa-shell'
error: build of '/nix/store/l3qv1nlaavy6llv0k4b2863fz3mdqbgz-shell.drv' failed
```

**Comment by worldofpeace** [comment]
> @worldofpeace I understood this, but when I try to use `nix-build` I get a confusing error message. Likely I'm doing something dumb...
> 
> ```
> $ nix-build -I nixpkgs=. -p "python3.withPackages(ps: with ps; [ pyvcd ])"
> these derivations will be built:
>   /nix/store/57x6z6360valwawys7dw387w2wac1sqw-python3-3.7.2-env.drv
>   /nix/store/l3qv1nlaavy6llv0k4b2863fz3mdqbgz-shell.drv
> building '/nix/store/57x6z6360valwawys7dw387w2wac1sqw-python3-3.7.2-env.drv'...
> created 245 symlinks in user environment
> building '/nix/store/l3qv1nlaavy6llv0k4b2863fz3mdqbgz-shell.drv'...
> builder for '/nix/store/l3qv1nlaavy6llv0k4b2863fz3mdqbgz-shell.drv' failed to produce output path '/nix/store/vx5y03618gf3hwn94fa2lyry3nmd40fa-shell'
> error: build of '/nix/store/l3qv1nlaavy6llv0k4b2863fz3mdqbgz-shell.drv' failed
> ```

I'd usually test by checking if the package builds for both python versions
```
nix-build -A python3.pkgs.pyvcd -A python.pkgs.pyvcd
```

Also noticed in your `nix-build` invocation that you specified the path to `nixpkgs` manually, and that isn't needed if you're current working directory is the checkout of nixpkgs.

And then using `nix-shell` to actually test if it works by using the result symlinks

```
nix-shell -p --pure ./result
nix-shell -p --pure ./result-2
```

You can also just do
```
nix-shell -I nixpkgs=. -p python.pkgs.pyvcd
```

**Pull request merged**